src/StrawberryfieldHydroponicsService.php

```diff
@@ -130,49 +130,165 @@ public function processQueue($name, $time = 120) {
+      $queue = $this->queueFactory->get($name, TRUE);
+      return $queue->numberOfItems();
+    }
+    else {
+      return 0;
+    }
+  }
+
+  /**
+  * Return Max Heartbeat delay expected
+  * It depends on processing type
+  */
+  public function getHearbeatMaxDelta() {
+    $config = $this->configFactory->get('strawberryfield.hydroponics_settings');
+    $processing_type = $config->get('processing_type') ? $config->get('processing_type') : "hydroponics";
+    $processing_monotime = $config->get('processing_monotime') ? $config->get('processing_monotime') : 60;
+    switch ($processing_type) {
+      case "hydroponics":
+        //if MONO then conservative, max queue process time + 5s
+        $heartbeat_max_delta = $processing_monotime + 5;
+        break;
+      case "hydroponicsmulti":
+        //if MULTI:
+        //due to child executed as process
+        //hearbeat interval is near to real time value
+        //so delta conservative could be heartbeat timer (3s) + 2s
+        $heartbeat_max_delta = 5;
+    }
+    return $heartbeat_max_delta;
+  }
 
   /**
    * Checks if Background Drush can run, and if so, sends it away.
    */
   public function run() {
     $config = $this->configFactory->get('strawberryfield.hydroponics_settings');
     if ($config->get('active')) {
+
+      $processing_type = $config->get('processing_type') ? $config->get('processing_type') : "mono";
+
       global $base_url;
       $site_path = \Drupal::service('site.path'); // e.g.: 'sites/default'
       $site_path = explode('/', $site_path);
       $site_name = $site_path[1];
       $queuerunner_pid = (int) \Drupal::state()->get('hydroponics.queurunner_last_pid', 0);
       $lastRunTime = intval(\Drupal::state()->get('hydroponics.heartbeat'));
       $currentTime = intval(\Drupal::time()->getRequestTime());
-      $running_posix = posix_kill($queuerunner_pid, 0);
-      if (!$running_posix || !$queuerunner_pid) {
+      $deltaTime = ($currentTime - $lastRunTime);
+
+      $heartbeat_max_delta = $this->getHearbeatMaxDelta();
+
+      $running_posix = FALSE;
+      if ($queuerunner_pid > 0) {
+        $running_posix = posix_kill($queuerunner_pid, 0);
+      }
+      //Normal stopped condition when all is ok
+      if (($deltaTime > $heartbeat_max_delta) && ($queuerunner_pid <= 0) && !$running_posix) {
         $this->logger->info('Hydroponics Service Not running, starting, time passed since last seen @time', [
-          '@time' => ($currentTime - $lastRunTime)]
+          '@time' => $deltaTime]
         );
         $path = $config->get('drush_path');
         if (empty($path)) {
           $path = '/var/www/html/vendor/drush/drush/drush';
         }
         $path = escapeshellcmd($path);
-        $cmd = $path.' archipelago:hydroponics --quiet --uri=' . $base_url;
+
+        //The parameter $processing_type is the right part of drush command
+        $cmd = $path.' archipelago:'.$processing_type.' --quiet --uri=' . $base_url;
```

Comment:
About this: I feel its better if we make the $processing_type the full drush command, thinking of future/external implementations where `archipelago` is not the default provider
This implies changing it here: https://github.com/esmero/strawberryfield/pull/138/files#diff-63ca552557d250f8063b3574fdfabb2b78b459992ca2aaacf04b9ce5ca2e48bfR198
from 
`hydroponics` to  `archipelago:hydroponics`

Reply:
done